orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -85,6 +85,35 @@ void testCompletion() {
     assertThat(result.getContent()).isNotEmpty();
   }
 
+  @Test
+  void testGrounding() {
+    stubFor(
+        post(anyUrl())
+            .willReturn(
+                aResponse()
+                    .withBodyFile("groundingResponse.json")
+                    .withHeader("Content-Type", "application/json")));
+    final var response = client.chatCompletion(prompt, config);
+    final var result = response.getOriginalResponse();
+    var llmChoice =
+        ((LLMModuleResultSynchronous) result.getOrchestrationResult()).getChoices().get(0);
+
+    assertThat(result.getModuleResults().getGrounding().getData().toString())
```

Comment:
(Minor/Comment)

I think AssertJ also has `.hasToString(` as replacement for `.toString()).isEqualTo(`

Reply:
I cannot compare the values directly here, as `getModuleResults().getGrounding().getData()` returns an `Object` so that I cannot access "grounding_query" or "grounding_result" via the generated code. This is why I compare it as a String.

Also, I use `.toString()).startsWith(...` now to not need to provide the complete rather long expected String.

Reply:
`Object` is somewhat misleading, actually this will always be a `Map<String, Object>`. (Ideally, our generator would be improved in that regard).

So you could cast to that and use `.get("grounding_query")`

Reply:
Casting this adds a warning, though. Is that a problem?

Reply:
yes, it's just a test assertion, so it's okay if this ever fails (which would mean we've changed our generator but didn't update tests)